### PR TITLE
Improve non-ambient class and function merge error

### LINKED ISSUE #32795: Improve error on non-ambient class and function merge
In #32584, we're allowing ambient classes and functions to merge, but non-ambient classes and functions still issue the same `Duplicate identifier` error. An improved, more specific, error would be [useful](https://github.com/microsoft/TypeScript/pull/32584#discussion_r312631137). Something mentioning that either the function can have an implementation or both need to be ambient.

### PR DESCRIPTION
**Pull request by austincummings**
Fixes #32795

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -26436,12 +26436,21 @@ namespace ts {
...
                 forEach(declarations, declaration => {
-                    addDuplicateDeclarationError(getNameOfDeclaration(declaration) || declaration, Diagnostics.Duplicate_identifier_0, symbolName(symbol), filter(declarations, d => d !== declaration));
+                    const diagnostic = declaration.kind === SyntaxKind.ClassDeclaration
+                            ? Diagnostics.Class_declaration_cannot_implement_overload_list_for_0
+                            : declaration.kind === SyntaxKind.FunctionDeclaration
+                                ? Diagnostics.Function_with_bodies_can_only_merge_with_classes_that_are_ambient
+                                : undefined;
+                    if (diagnostic) {
+                        addRelatedInfo(
+                            error(getNameOfDeclaration(declaration) || declaration, diagnostic, symbolName(symbol)),
```
Comment (austincummings) [inline src/compiler/checker.ts]: Is it safe to provide `symbolName(symbol)` even if the diagnostic doesn't specify a parameter?

File: src/compiler/diagnosticMessages.json [hunk 2]
```diff
@@ -2693,6 +2693,14 @@
         "category": "Error",
         "code": 2773
     },
+    "Class declaration cannot implement overload list for '{0}'.": {
+        "category": "Error",
+        "code": 2774
+    },
+    "Function with bodies can only merge with classes that are ambient.": {
```
Comment (austincummings) [inline src/compiler/diagnosticMessages.json]: Is **bodies** right? Or can there only be one function body?

**Comment by orta** [comment]
Poke ^

**Comment by orta** [comment]
Hi @austincummings - this PR didn't have 'allow maintainers to make changes' checked, and so when we migrated TypeScript's primary branch from 'master' to 'main' this PR couldn't have its branch re-targeted. 

It looks like most of my changes were cosmetics, so once I've got my current PR up and running, I'll replicate this PR with my changes included and get it reviewed 👍🏻

**Pull request closed without merging**